src/transformers/pipelines/image_to_text.py

```diff
@@ -84,23 +90,70 @@ def __call__(self, images: Union[str, List[str], "Image.Image", List["Image.Imag
                 - An image loaded in PIL directly
 
                 The pipeline accepts either a single image or a batch of images.
+            texts (`str`, `List[str]`, *optional*):
+                The pipeline handles conditional generation. If `texts` is provided, the model will generate the image
+                caption conditioned on the text.
 
             max_new_tokens (`int`, *optional*):
                 The amount of maximum tokens to generate. By default it will use `generate` default.
 
-            generate_kwargs (`Dict`, *optional*):
-                Pass it to send all of these arguments directly to `generate` allowing full control of this function.
+            kwargs (`Dict`, *optional*):
+                The kwargs will contain the preprocessing kwargs (including kwargs related to text input) as well as
+                the generate kwargs that will send all of these arguments directly to `generate` allowing full control
+                of this function.
 
         Return:
             A list or a list of list of `dict`: Each result comes as a dictionary with the following key:
 
             - **generated_text** (`str`) -- The generated text.
         """
-        return super().__call__(images, **kwargs)
-
-    def preprocess(self, image):
-        image = load_image(image)
-        model_inputs = self.image_processor(images=image, return_tensors=self.framework)
+        model_inputs = None
+        if isinstance(images, list) and texts is not None and isinstance(texts, list):
+            model_inputs = [{"images": image, "texts": texts} for image, texts in zip(images, texts)]
+        elif isinstance(images, list) and texts is not None:
+            # same text on all images
+            model_inputs = [{"images": image, "texts": texts} for image in images]
+        elif isinstance(images, list) and texts is None:
+            # no text
+            model_inputs = [{"images": image} for image in images]
+        elif not isinstance(images, list) and texts is None:
+            # classic input with only images
+            model_inputs = images
+        elif not isinstance(images, list) and texts is not None:
+            # classic input with images and text
+            model_inputs = {"images": images, "texts": texts}
```

Comment:
This is intrinsically bad code.

Any code looking like this in other pipelines was added for legacy purposes and being backward compatible.
Adding this type of code is inherently wrong IMO.

Imagine you are not allowed to inspect anything coming your way. You need to trust the base class to unpack everything for you, and `preprocess` will receive a single image. That's it.
No preprocessing can happen here.

That means text cannot be an actual argument of the pipeline (as mentioned in the previous comment, the only potentially satisfactory path, is more like a glorified parameter being 100% optional). Sending different texts on 1 given call should be purely disallowed IMO (because of all the hoops you have to go through here).

And I say that without even starting to mention all the missing checks everywhere for sizes and the fact that the output of this path is not even a single type. (There's a the very least a dict vs list of dicts and most likely an entire array of potential nullables everywhere.)

Reply:
Thanks, after thinking about it that make sense. I had to come up with these hacks to deal with different configurations (mainly, multiple images + text), which should not be supported as you are suggesting, I proposed some changes in 4f08c05 & [ed4a69d](https://github.com/huggingface/transformers/pull/22423/commits/ed4a69d0eba87620901b459df41dd718931d324c)